File: F1GameDataParser/ViewModels/TimingTower/TimingTower.cs
using F1GameDataParser.Enums;

namespace F1GameDataParser.ViewModels.TimingTower
{
    public class TimingTower
    {
        public GameYear GameYear { get; set; }
        public bool IsRaceSession { get; set; }
        public bool IsSessionFinished { get; set; }
        public byte CurrentLap { get; set; }
        public byte TotalLaps { get; set; }

        public SafetyCarType SafetyCarStatus { get; set; }
        public IEnumerable<bool> SectorYellowFlags { get; set; }
        public AdditionalInfoType ShowAdditionalInfo { get; set; }
        public string? SessionTimeLeft { get; set; }

        public IEnumerable<DriverTimingDetails> DriverTimingDetails { get; set; }
        public byte SpectatorCarIdx { get; set; }
    }
}

File: F1BroadcastTools.Client/src/app/shared/models/TimingTower.ts
import { DriverStatus, GameYear, ResultStatus, SafetyCarStatus, Team } from "./Enumerations";
import { TeamDetails } from "./team.model";

export interface TimingTower {
    gameYear: GameYear;
    isRaceSession: boolean;
    isSessionFinished: boolean;
    currentLap: number;
    totalLaps: number;

    safetyCarStatus: SafetyCarStatus;
    sectorYellowFlags: boolean[];
    showAdditionalInfo: number;
    sessionTimeLeft: string | null;

    driverTimingDetails: DriverTimingDetails[];
    spectatorCarIdx: number;
}

export interface DriverTimingDetails {
    vehicleIdx: number;
    position: number;
    teamId: Team;
    teamDetails?: TeamDetails;
    name: string;
    tyreAge: number;
    visualTyreCompound: string;
    gap: string;
    resultStatus: ResultStatus;
    driverStatus: DriverStatus;
    penalties: number;
    warnings: number;
    hasFastestLap: boolean;
    isInPits: boolean;
    numPitStops: number;
    positionsGained: number;
}

export enum Sector
{
    Sector1,
    Sector2,
    Sector3
}